extensions/jolt/deployment/src/main/java/org/apache/camel/quarkus/component/jolt/deployment/JoltProcessor.java

```diff
@@ -36,5 +36,15 @@ void registerReflectiveClasses(BuildProducer<ReflectiveClassBuildItem> producer)
         ChainrEntry.STOCK_TRANSFORMS.values().stream().forEach(c -> {
             producer.produce(new ReflectiveClassBuildItem(false, false, c));
         });
+
+        // A JOLT Defaultr transformation spec is a JSON content and it needs to be serialized at some point.
+        // As such, we need to register all JSON base types and super types for serialization.
+        producer.produce(new ReflectiveClassBuildItem(false, false, false, true, java.lang.Boolean.class.getName()));
+        producer.produce(new ReflectiveClassBuildItem(false, false, false, true, java.lang.Number.class.getName()));
+        producer.produce(new ReflectiveClassBuildItem(false, false, false, true, java.lang.Double.class.getName()));
+        producer.produce(new ReflectiveClassBuildItem(false, false, false, true, java.lang.Integer.class.getName()));
+        producer.produce(new ReflectiveClassBuildItem(false, false, false, true, java.lang.String.class.getName()));
+        producer.produce(new ReflectiveClassBuildItem(false, false, false, true, java.util.AbstractList.class.getName()));
+        producer.produce(new ReflectiveClassBuildItem(false, false, false, true, java.util.ArrayList.class.getName()));
```

Comment:
There is CamelSerializationBuildItem and https://github.com/apache/camel-quarkus/blob/main/extensions-core/core/deployment/src/main/java/org/apache/camel/quarkus/core/deployment/CamelSerializationProcessor.java#L42-L71 that perhaps do 99% of what we need here?
As far as I can see only ArrayList is missing in CamelSerializationProcessor. Maybe we could add it there and save a couple of lines here?

Reply:
Make sense to consider JSON types as common for camel. I've pushed an update in this direction.